extensions/elasticsearch/elasticsearch-5/src/main/java/com/hazelcast/jet/elastic/impl/ElasticSourcePMetaSupplier.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2020 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.elastic.impl;
+
+import com.hazelcast.cluster.Address;
+import com.hazelcast.cluster.Member;
+import com.hazelcast.jet.JetException;
+import com.hazelcast.jet.core.ProcessorMetaSupplier;
+import com.hazelcast.jet.core.ProcessorSupplier;
+import com.hazelcast.jet.core.processor.Processors;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestHighLevelClient;
+
+import javax.annotation.Nonnull;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.nCopies;
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toSet;
+
+public class ElasticSourcePMetaSupplier<T> implements ProcessorMetaSupplier {
+
+    private static final long serialVersionUID = 1L;
+    private static final int DEFAULT_LOCAL_PARALLELISM = 2;
+
+    @Nonnull
+    private final ElasticSourceConfiguration<T> configuration;
+
+    private transient Map<Address, List<Shard>> assignedShards;
+    private transient Address ownerAddress;
+
+    public ElasticSourcePMetaSupplier(@Nonnull ElasticSourceConfiguration<T> configuration) {
+        this.configuration = configuration;
+    }
+
+    @Override
+    public int preferredLocalParallelism() {
+        if (configuration.isCoLocatedReadingEnabled() || configuration.isSlicingEnabled()) {
+            return DEFAULT_LOCAL_PARALLELISM;
+        } else {
+            return 1;
+        }
+    }
+
+    @Override
+    public void init(@Nonnull Context context) throws Exception {
+        RestHighLevelClient client = configuration.clientFn().get();
```

Comment:
I see that we have this obtaining RestClient via reflection at processor too. Maybe we can have it as a static utility method, in ElasticCatClient for example.

Reply:
Done.